internal/client/client_connection.go

```diff
@@ -74,9 +74,9 @@ func (c *client) sendToBroker(MessageID uint16, topic string, qos byte, retained
 	}
 }
 
-func (c *client) login(username, password string) bool {
+func (c *client) login(client_id, username, password string) bool {
 	for _, conf := range c.configs {
-		if conf.Login == username && conf.Password == password {
+		if conf.ClientID == client_id && conf.Login == username && conf.Password == password {
```

Comment:
ClientId should not be used in authentication. It doesn't correspond MQTT specs https://www.hivemq.com/blog/mqtt-essentials-part-3-client-broker-connection-establishment/
Why do you need it?